python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,266 @@
+            else:
+                break
+    return pd.DataFrame(rows)
+
+
+def generate_data(args):
+    """
+    Generate data with the specified parameters
+    - n_classes: Number of classes in the datasets. Should be 2 (clicked, not clicked)
+    - n_samples: Number of samples to generate
+    - n_features: Number of features per sample
+    - n_redundant: The number of redundant features
+    - n_informative: The number of informative (Non-redundant) features
+    - n_clusters_per_class: The number of clusters per class.
+    - flip_y: The fraction of samples whose class is assigned randomly. Larger
+        values introduce noise in the labels and make the classification
+        task harder.
+    - class_sep: Larger values spread out the clusters/classes and make the classification task easier.
+    - seed: Random seed
+    - n_trials: Number of trials of data generation to obtain a linearly separable data
+    Returns: the best seprabale data set by a Perceptorn.
+    """
+    separable = False
+    best_accuracy = 0
+    best_accuracy_data = None
+    random.seed(args.seed)
+    seeds = list(range(args.n_trials))
+    random.shuffle(seeds)
+    trials = len(seeds)-1
+    while not separable and trials > 0:
+        samples = datasets.make_classification(n_classes=args.n_classes, n_samples=args.n_samples, n_features=args.n_features,
+                                               n_redundant=args.n_redundant, n_informative=args.n_informative,
+                                               n_clusters_per_class=args.n_clusters_per_class, flip_y=args.flip_y,
+                                               class_sep=args.class_sep, random_state=seeds[trials])
+        p = Perceptron(fit_intercept=True, max_iter=1000, verbose=0, random_state=args.seed, validation_fraction=0.0001)
+        p.fit(samples[0], samples[1])
+        acc = p.score(samples[0], samples[1])
+        if acc > best_accuracy:
+            best_accuracy = acc
+            best_accuracy_data = samples
+        if acc == 1.0:
+            separable = True
+        trials -= 1
+    return best_accuracy_data[0], best_accuracy_data[1]
+
+
+def calculate_MRR(prediction, y):
+    """
+    Calculates the MRR given the predicted scores (prediction) and the actual clicks (y).
+    """
+    if sum(y) < 1:
+        return 0
+    pred = prediction
+    clickedDocIdxs = [i for i, x in enumerate(y) if x == 1.0]
+    mrr = 0
+    sortedPred = sorted(pred)[::-1]
+    for clickedDocIdx in clickedDocIdxs:
+        clickedDocScore = pred[clickedDocIdx]
+        # ties are resolved with the worst case (i.e. the correct doc will have the lowest place among other docs with same score)
+        equalScoreCount = sortedPred.count(clickedDocScore)
+        predictedRank = sortedPred.index(clickedDocScore) + equalScoreCount - 1.0
+        mrr += 1.0 / (predictedRank + 1.0)
+    mrr = mrr / len(clickedDocIdxs)
+    return mrr
+
+
+def check_MRR(df, model, featureCount=2):
```

Comment:
@mohazahran [Minor] Why are you switching to camel case for variable names here? Can we keep it consistent?